c/uca/sifter/unisifeggs.cpp

```diff
@@ -0,0 +1,433 @@
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <ranges>
+#include <string>
+#include <string_view>
+#include <vector>
+
+class CheckFailure {
+ public:
+  ~CheckFailure() {
+    std::terminate();
+  }
+  void operator&(std::ostream&) {}
+};
+
+#define CHECK(condition) \
+  (condition)            \
+      ? (void)0          \
+      : CheckFailure() & std::cerr << ("Check failed: " #condition "\n")
```

Comment:
```suggestion
#define CHECK(condition) \
  ((condition)            \
      ? (void)0          \
      : CheckFailure() & std::cerr << ("Check failed: " #condition "\n"))
```
... so that, if someone accidentally puts this into a larger expression, we don't get funny operator precedence effects.

Reply:
No, this is intentional so you can output additional context into it (see the glog macros).